test/unit/domain/shared/amount-calculator.spec.ts

```diff
@@ -57,92 +57,97 @@ describe("AmountCalculator", () => {
     const divisorBound = 50n
     const quotient = 2n
 
-    describe("by rounding normally", () => {
-      it("no rounding if remainder is 0", () => {
-        const result = calc.divRound({ amount: pivot, currency }, divisor)
-        expect(result.amount).toEqual(quotient)
-        expect(result.currency).toBe(currency)
+    const describeCases = [
+      {
+        type: "normal",
+        div: calc.divRound,
+        up: "up",
+        down: "down",
+      },
+      {
+        type: "floor",
+        div: calc.divFloor,
+        up: "down",
+        down: "down",
+      },
+      {
+        type: "ceil",
+        div: calc.divCeil,
+        up: "up",
+        down: "up",
+      },
+    ]
+    const itCases = ({ up, down }) => [
+      {
+        title: `no rounding if remainder is 0`,
+        amount: pivot,
+        result: {
+          normal: quotient,
+          floor: quotient,
+          ceil: quotient,
+        },
+      },
+      {
+        title: `rounds ${down} if remainder is just above pivot`,
+        amount: pivot + 1n,
+        result: {
+          normal: quotient,
+          floor: quotient,
+          ceil: quotient + 1n,
+        },
+      },
+      {
+        title: `rounds ${up} if remainder is just below pivot`,
+        amount: pivot - 1n,
+        result: {
+          normal: quotient,
+          floor: quotient - 1n,
+          ceil: quotient,
+        },
+      },
+      {
+        title: `rounds ${down} if remainder is just below half of the divisor`,
+        amount: pivot + (divisorBound - 1n),
+        result: {
+          normal: quotient,
+          floor: quotient,
+          ceil: quotient + 1n,
+        },
+      },
+      {
+        title: `rounds ${up} if remainder is just above half of the divisor`,
+        amount: pivot + (divisorBound + 1n),
+        result: {
+          normal: quotient + 1n,
+          floor: quotient,
+          ceil: quotient + 1n,
+        },
+      },
+      {
+        title: `rounds ${down} if remainder is half of the divisor`,
+        amount: pivot + divisorBound,
+        result: {
+          normal: quotient,
+          floor: quotient,
+          ceil: quotient + 1n,
+        },
+      },
+    ]
+
+    for (const describeGroup of describeCases) {
+      describe(`by rounding to ${describeGroup.type}`, () => {
```

Comment:
uhmm not sure how jest behaves with this for syntax so I think is better to use test.each https://jestjs.io/docs/api#testeachtablename-fn-timeout

Reply:
It works well in local testing so far.

I tried `test.each` at first for this, but the test titles are also dependent on the `table` and `test.each` only gets table items in its 2nd `fn` argument.